Refactor 'df' usage

djmitche opened the pull request:
This inlcudes more tests and refactors the use of 'df' to:
 * parse the header and use it to find various fields, rather than
   assuming things contain no ' ' characters
 * operate in `filesystem.go` since it's almost completely
   non-OS-specific now.

sgnn7 commented on filesystem/filesystem.go:
```diff
@@ -49,22 +57,52 @@ func getFileSystemInfo() (interface{}, error) {
+			field, dfCommand, strings.Join(dfOptions, " "))
+	}
+
+	kbsizeOffset := strings.Index(hdr, "1K-blocks")
```
I don't think we can reliably key off from the header. The [strings in the command](https://github.com/coreutils/coreutils/blob/master/src/df.c#L180-L183) are [wrapped in `N_()` macro](https://github.com/coreutils/coreutils/blob/master/src/df.c#L180-L183) and [some text](https://github.com/coreutils/coreutils/blob/master/src/df.c#L620-L632) is wrapped in [`_()` macro](https://www.freedesktop.org/software/gstreamer-sdk/data/docs/latest/glib/glib-I18N.html#gettext-macro) which [might be implying](https://www.freedesktop.org/software/gstreamer-sdk/data/docs/latest/glib/glib-I18N.html#N-:CAPS) that there may be i10n around them.

Because of this, I wonder if we should ensure we use `-p` and/or `--output` and not depend on headers to parse this data. As a follow-up, do we maybe have some newer golang primitives for this too so that we don't have to deal with CLI?

djmitche replied:
Ah, good point.  I think we'd need to set some localization variables like `LC_ALL=C` to ensure we get the un-translated output.

Using `-P` or `--output` would have the same problem -- different headers, but still variable-width.  And switching to gopsutil might be a breaking change, in terms of which filesystems are included or excluded.  We really don't have enough data to tell.

sgnn7 reviewed (commented):
@djmitche Left a comment around parsing approach but overall I like the idea of merging the two implementations

djmitche commented:
OK, I added `LC_ALL=C` to the `df` invocation to ensure we're not getting translated headers.

I also removed the different dfOptions on different platforms.  Both Mac and Linux can use `df -k -l`.
